### ISSUE TITLE
Remove programmatic pin map creation API from nidigital

### ISSUE THREAD

**Issue by marcoskirsch** [ni/nimi-python#1124]
Programmatic pin map creation is a very niche need that needs a lot of functions and testing. The vast majority of users load pin maps that they create in the PBD / TSM pin map editors or using a text editor.

We should not expose such niche functionality unless there is a very compelling reason to do so.

**Comment by sbethur** [comment]
TL;DR: Yes, we can remove programmatic pin map creation API.

Discussed this with the Digital driver team..
- We are not breaking any use-case if we don't provide the programmatic API
- We are not aware of any customer workflow that requires programmatic API
- Programmatic pinmap API was added in Digital 0.9 as a stop-gap before the loading of pinmap file was made available in 1.0
- It's always easier to add than remove

Given these points, we can remove the programmatic pinmap creation API.

For the sake of completeness, we also discussed about other similar "low-level" API for configuring levels and timesets. We want to keep these API, even though there is load-file way, because it's common for levels/timeset values to be changed during a test program, i.e. Load a set of levels/timeset values from file and then modify one of more values during test execution as required.

**Comment by marcoskirsch** [comment]
So we can close #1186 ?

**Comment by sbethur** [comment]
> So we can close #1186 ?

Yes

**Comment by markusroeleke** [comment]
We at Bosch want to use the programmatic API to generate the PinMap. I now have to write an generator to generate the PinMap in order to load it to the card. We want to reconfigure the PXI6570 dynamically during runtime of the test program. We have a couple of cards here. Please add this functionality again.

**Comment by sbethur** [comment]
@markusroeleke, First off, thanks for using the pre-release nidigital API and giving us feedback before we finalize the API. We want to understand your usecase better and have a few questions:

1. Can you please elaborate on how you're dynamically reconfiguring 6570 during runtime?
    - Are you closing and reopening the session to load a new pinmap or are you calling `reset()`?
    - Does the runtime dynamic reconfiguration also involve changes to the physical test setup like changing the loadboard/DUT, rewiring the connections, etc.?
2. How many pinmaps are being used in the test program and how different are they?
3. What kind of DUTs are being tested?
4. Is the test program for characterization or production test?
5. Are you using TestStand Semiconductor Module?

**Comment by marcoskirsch** [comment]
@markusroeleke please open a new issue and add as much context as possible. Thanks!
